src/linux_ip_socket/linux_ip_socket.h

```diff
@@ -0,0 +1,180 @@
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LINUX_IP_SOCKET_H
+#define LINUX_IP_SOCKET_H
+
+/**
+ * @file     linux_ip_socket.h
+ * @brief    Driver for TASTE with uses TCP/IP for communication.
+ *
+ */
+
+#include <cstddef>
+#include <cstdint>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <poll.h>
+
+#include <Thread.h>
+#include <system_spec.h>
+
+#include <drivers_config.h>
+
+extern "C"
+{
+#include <Broker.h>
+}
+
+/**
+ * @brief Structure for driver internal data.
+ *
+ * This structure is allocated by runtime and the pointer is passed to all driver functions.
+ * The name of this structure shall match driver definition from ocarina_components.aadl
+ * and has suffix '_private_data'.
+ */
+class linux_ip_socket_private_data final
+{
+  public:
+    /**
+     * @brief  Constructor.
+     *
+     * Construct empty object, which needs to be initialized using @link linux_ip_socket_private_data::init
+     * before usage.
+     */
+    linux_ip_socket_private_data();
+
+    /**
+     * @brief Initialize driver.
+     *
+     * Driver needs to be initialized before start.
+     *
+     * @param bus_id         Identifier of the bus, which is used by driver
+     * @param device_id      Identifier of the device
+     * @param device_configuration Configuration of device
+     * @param remote_device_configuration Configuration of remote device
+     */
+    void driver_init(const SystemBus bus_id,
+                     const SystemDevice device_id,
+                     const Socket_IP_Conf_T* const device_configuration,
+                     const Socket_IP_Conf_T* const remote_device_configuration);
+    /**
+     * @brief Receive data from remote partitions.
+     *
+     * This function receives data from remote partition and sends it to the Broker.
+     */
+    void driver_poll();
+    /**
+     * @brief send data to remote partition.
+     *
+     * @param data           The Buffer which data to send to connected remote partition
+     * @param length         The size of the buffer
+     */
+    void driver_send(const uint8_t* data, const size_t length);
+
+  private:
+    enum PacketParseState
+    {
+        STATE_WAIT,
+        STATE_DATA_BYTE,
+        STATE_ESCAPE_BYTE,
+    };
+
+    static constexpr int DRIVER_THREAD_PRIORITY = 1;
+    static constexpr int DRIVER_THREAD_STACK_SIZE = 65536;
```

Comment:
Why 64k?

Reply:
Good question. 
The default stack size in TASTE is 64k. This is enaugh for this driver implementation.